src/mo_frontend/source_lexer.mll

```diff
@@ -91,13 +91,15 @@ let utf8 = ascii | utf8enc
 let utf8_no_nl = ascii_no_nl | utf8enc
 
 let byte = '\\'hexdigit hexdigit
-let escape = ['n''r''t''\\''\'''\"']
+let escape = ['t''\\''\'''\"']
 let character =
     [^'"''\\''\x00'-'\x1f''\x7f'-'\xff']
   | utf8enc
   | byte
   | '\\'escape
   | "\\u{" hexnum '}'
+  | '\n'
```

Comment:
Simply throwing in newline into a token is broken. Every newline requires a call to Lexing.new_line, otherwise line number tracking for errors and warnings will be off.